src/frontend/api.ts

```diff
@@ -195,13 +210,13 @@ export function Api({
     status = value;
 
     if (status === STATUS.LISTENING) {
-      while (pending.length) pending.shift().resolve();
+      while (pending.length) pending.shift()?.resolve();
```

Comment:
it's there because the returned result of `shift` could be undefined based on `pending`'s type def

```ts
const pending: Array<{
  resolve: () => any;
  reject: (err: Error) => void;
}> = [];
```

Reply:
wait my explanation makes less sense lol. what you described is exactly the reasoning actually 😆